.github/workflows/node.js.yml

```diff
@@ -31,22 +30,53 @@ jobs:
           node-version-file: .tool-versions
 
       - name: 📦 Install dependencies
-        run: yarn install
+        run: yarn install --immutable --immutable-cache
 
       - name: 💅 Prettier check
         run: yarn prettier:check
 
       - name: 💨 Lint check
         run: yarn lint
 
+  test:
+    name: ✅ Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Node.js setup
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .tool-versions
+
+      - name: 📦 Install dependencies
+        run: yarn install --immutable --immutable-cache
+
       - name: ✅ Run Tests
         run: yarn test --no-watch
 
+  build:
+    name: 🏗️ Build app
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Node.js setup
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .tool-versions
+
+      - name: 📦 Install dependencies
+        run: yarn install --immutable --immutable-cache
+
       - name: 🏗️ Build app
         run: yarn build
 
   deploy:
-    name: Deploy
+    name: 🚀 Deploy app
```

Comment:
do we want to make Deploy dependent on all the other jobs, or just on `build`?

Reply:
Good call